cypress/integration/tip.test.ts

```diff
@@ -1,80 +1,98 @@
 
 const stringToCurrency = (amount: string, currency: Currency) =>
   currency instanceof Token ? new TokenAmount(currency, JSBI.BigInt(amount)) : CurrencyAmount.ether(JSBI.BigInt(amount))
 
 // Assumptions:
 // RINKEBY
-// SELL: ETH
-// BUY: DAI
-const DAI = '0xc7AD46e0b8a400Bb3C915120d284AafbA8fc4735'
-const INPUT_TOKEN = 'ETH'
-const OUTPUT_TOKEN = DAI
-
-const INPUT_AMOUNT = '10'
-const FEE_QUERY = `https://protocol-rinkeby.dev.gnosisdev.com/api/v1/tokens/${checkIfEther(INPUT_TOKEN, 4)}/fee`
-const SWAP_URL = `/swap?inputCurrency=${INPUT_TOKEN}&outputCurrency=${OUTPUT_TOKEN}`
-
-describe('Swap::Trade::Tip', () => {
-  beforeEach(() => {
-    cy.visit(SWAP_URL)
-  })
+const TYPED_AMOUNT = '12'
+// DAI
+const DAI_ADDRESS = '0xc7AD46e0b8a400Bb3C915120d284AafbA8fc4735'
+const OUTPUT_TOKEN = new Token(4, DAI_ADDRESS, 18, 'DAI', 'DAI Stablecoin')
 
-  it('Fee query available', () => {
-    cy.request(FEE_QUERY).then(response => {
-      expect(response).to.have.property('body')
-      expect(response.body).to.have.property('minimalFee') // true
-      expect(response.body).to.have.property('feeRatio') // true
-      expect(response.body).to.have.property('expirationDate') // true
-    })
+const FEE_QUERY = `https://protocol-rinkeby.dev.gnosisdev.com/api/v1/tokens/${WETH[4].address}/fee`
+const SWAP_URL = '/swap'
+
+describe('[EXACT-IN::SELL] :: ETH/DAI', () => {
+  before(() => {
+    cy.visit(SWAP_URL)
   })
 
-  it('Swap ETH for DAI and applies fee', () => {
+  it('[UI] Selects DAI token', () => {
     // Open currency selector
     cy.get('#swap-currency-output .open-currency-select-button').click()
-    cy.get(`.token-item-${DAI}`).should(`be.visible`)
+    cy.get(`.token-item-${OUTPUT_TOKEN.address}`).should(`be.visible`)
     // select DAI
-    cy.get(`.token-item-${DAI}`).click({ force: true })
+    cy.get(`.token-item-${OUTPUT_TOKEN.address}`).click({ force: true })
     cy.get('#swap-currency-input .token-amount-input').should('be.visible')
     // input the cached input amount (above)
-    cy.get('#swap-currency-input .token-amount-input').type(INPUT_AMOUNT, { force: true, delay: 200 })
+    cy.get('#swap-currency-input .token-amount-input').type(TYPED_AMOUNT, { force: true, delay: 200 })
     cy.get('#swap-currency-output .token-amount-input').should('not.equal', '')
+  })
+
+  context('[Trade + Fee Logic Tests] :: ETH/DAI', () => {
+    beforeEach(() => {
+      // set aliases in context to reuse
+      // start fee request and alias as @fee
+      cy.request(FEE_QUERY)
+        .its('body')
+        .as('fee')
+
+      // Trade object and alias as @trade
+      cy.window()
+        .as('win')
+        .its('trade')
+        .as('trade')
+
+      cy.get('#swap-currency-output .token-amount-input').as('output')
+
+      // Clear the storage (old redux state etc)
+      cy.get<Window>('@win').then(window => window.localStorage.clear())
+    })
+
+    it('[TRADE] Trade properties are correct', () => {
```

Comment:
I find very strange this test. It seems you want to try the SDK using integration test. 
Shouldn't we test by checking the output in the UI?

![image](https://user-images.githubusercontent.com/2352112/106018526-e614a780-60c1-11eb-9ecb-1c682fafd1cf.png)